DotNetMSBuild alias generates extra verbosity argument

### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake Frosting

### Cake version

5.0.0

### Operating system

Windows

### Operating system architecture

x64

### CI Server

_No response_

### What are you seeing?

When specifying the `Verbosity` property on the `DotNetMsbuildSettings` class the dotnet msbuild tool outputs an error:
```
MSBUILD : error MSB1016: Specify the verbosity level.
Full command line: 'C:\Program Files\dotnet\sdk\8.0.404\MSBuild.dll -maxcpucount -verbosity:m test.proj --verbosity detailed -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\8.0.404\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\8.0.404\dotnet.dll'
```

### What is expected?

The `dotnet msbuild` command should not include the `--verbosity detailed` that is included with most of the other DotNet* aliases.

### Steps to Reproduce

**test.proj**
```xml
<Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
  <Target Name="Echo">
    <Message Importance="high" Text="High Message" />
    <Message Importance="normal" Text="Normal Message" />
    <Message Importance="low" Text="Low Message" />
  </Target>
</Project>
```

**Program.cs**
```
using Cake.Common.Tools.DotNet;
using Cake.Common.Tools.DotNet.MSBuild;
using Cake.Core;
using Cake.Frosting;

public static class Program
{
    public static int Main(string[] args)
    {
        return new CakeHost()
            .Run(args);
    }
}

[TaskName("Hello")]
public sealed class HelloTask : FrostingTask
{
	public override void Run(ICakeContext context)
	{
		context.DotNetMSBuild("test.proj", new DotNetMSBuildSettings()
		{
			Verbosity = DotNetVerbosity.Detailed
		});
	}
}
```

**build.csproj**
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <RunWorkingDirectory>$(MSBuildProjectDirectory)</RunWorkingDirectory>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Cake.Frosting" Version="5.0.0" />
  </ItemGroup>
</Project>
```

Then run:
```
dotnet run --project .\build.csproj --target Hello
```

### Output log

```
C:\Program Files\dotnet\sdk\8.0.404\MSBuild.dll -nologo -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\8.0.404\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\8.0.404\dotnet.dll -maxcpucount -restore -verbosity:m -verbosity:diagnostic .\Build.csproj
Build started 1/30/2025 11:08:11.
Environment at start of build:
MSBuildSDKsPath = C:\Program Files\dotnet\sdk\8.0.404\Sdks
MSBUILDUSESERVER = 0
MSBuildLoadMicrosoftTargetsReadOnly = true
MSBUILDFAILONDRIVEENUMERATINGWILDCARD = 1
DOTNET_CLI_TELEMETRY_SESSIONID = 99d966f0-a5f3-4ce3-b75e-d7afa6a04644
DOTNET_HOST_PATH = C:\Program Files\dotnet\dotnet.exe
MSBuildExtensionsPath = C:\Program Files\dotnet\sdk\8.0.404\

11:08:11.436     0>Process = "C:\Program Files\dotnet\dotnet.exe"
                   MSBuild executable path = "C:\Program Files\dotnet\sdk\8.0.404\MSBuild.dll"
                   Command line arguments = "C:\Program Files\dotnet\sdk\8.0.404\MSBuild.dll -maxcpucount -verbosity:m -nologo -restore .\Build.csproj -nologo -verbosity:diagnostic -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\8.0.404\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\8.0.404\dotnet.dll"
                   MSBuild version = "17.11.9+a69bbaaf5"
                   Based on the Windows registry key LongPathsEnabled, the LongPaths feature is enabled.

//
// snip - bunch of msbuild output...
//

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.93

========================================
Hello
========================================
MSBuild version 17.11.9+a69bbaaf5 for .NET
MSBUILD : error MSB1016: Specify the verbosity level.
    Full command line: 'C:\Program Files\dotnet\sdk\8.0.404\MSBuild.dll -maxcpucount -verbosity:m test.proj --verbosity detailed -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\8.0.404\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\8.0.404\dotnet.dll'
  Switches appended by response files:
Switch: --verbosity

For switch syntax, type "MSBuild -help"
An error occurred when executing task 'Hello'.
Error: .NET CLI: Process returned an error (exit code 1).
```

--- a/src/Cake.Common.Tests/Unit/Tools/DotNet/MSBuild/DotNetMSBuildBuilderTests.cs
+++ b/src/Cake.Common.Tests/Unit/Tools/DotNet/MSBuild/DotNetMSBuildBuilderTests.cs
@@ -1,4 +1,4 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
@@ -909,6 +909,26 @@ namespace Cake.Common.Tests.Unit.Tools.DotNet.MSBuild
 
                 // Then
                 Assert.Equal("--diagnostics msbuild", result.Args);
+            }
+
+            [Theory]
+            [InlineData(DotNetVerbosity.Quiet, "/verbosity:quiet")]
+            [InlineData(DotNetVerbosity.Minimal, "/verbosity:minimal")]
+            [InlineData(DotNetVerbosity.Normal, "/verbosity:normal")]
+            [InlineData(DotNetVerbosity.Detailed, "/verbosity:detailed")]
+            [InlineData(DotNetVerbosity.Diagnostic, "/verbosity:diagnostic")]
+            public void Should_Use_MSBuild_Verbosity_Not_DotNet_Verbosity_When_Verbosity_Is_Specified(DotNetVerbosity verbosity, string expectedMsBuildVerbosity)
+            {
+                // Given: DotNetMSBuildSettings.Verbosity (not ConsoleLoggerSettings) causes MSB1016 if passed as dotnet --verbosity
+                var fixture = new DotNetMSBuildBuilderFixture();
+                fixture.Settings.Verbosity = verbosity;
+
+                // When
+                var result = fixture.Run();
+
+                // Then: must use MSBuild /verbosity switch, not dotnet --verbosity (see https://github.com/cake-build/cake/issues/4456)
+                Assert.Contains(expectedMsBuildVerbosity, result.Args);
+                Assert.DoesNotContain("--verbosity", result.Args);
             }
 
             [Fact]

--- a/src/Cake.Common/Tools/DotNet/MSBuild/DotNetMSBuildBuilder.cs
+++ b/src/Cake.Common/Tools/DotNet/MSBuild/DotNetMSBuildBuilder.cs
@@ -1,4 +1,4 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
@@ -13,6 +13,10 @@ namespace Cake.Common.Tools.DotNet.MSBuild
     /// <summary>
     /// .NET Core project builder.
     /// </summary>
+    /// <remarks>
+    /// Verbosity is passed as MSBuild /verbosity (not dotnet --verbosity) because dotnet msbuild forwards
+    /// arguments to MSBuild, which does not accept the dotnet CLI --verbosity form and fails with MSB1016.
+    /// </remarks>
     public sealed class DotNetMSBuildBuilder : DotNetTool<DotNetMSBuildSettings>
     {
         private readonly ICakeEnvironment _environment;
@@ -39,13 +43,18 @@ namespace Cake.Common.Tools.DotNet.MSBuild
         /// <param name="projectOrDirectory">The target project path.</param>
         /// <param name="settings">The settings.</param>
         /// <param name="standardOutputAction">The action to invoke with the standard output.</param>
+        /// <remarks>
+        /// Calls Run directly so that dotnet --verbosity is not appended (it would cause MSB1016);
+        /// verbosity is passed as MSBuild /verbosity in the arguments instead.
+        /// </remarks>
         public void Build(string projectOrDirectory, DotNetMSBuildSettings settings, Action<IEnumerable<string>> standardOutputAction)
         {
             ArgumentNullException.ThrowIfNull(settings);
 
-            RunCommand(
+            var arguments = GetArguments(projectOrDirectory, settings);
+            Run(
                 settings,
-                GetArguments(projectOrDirectory, settings),
+                arguments,
                 standardOutputAction == null ? null : new ProcessSettings { RedirectStandardOutput = true },
                 standardOutputAction == null ? null : new Action<IProcess>(process => standardOutputAction(process.GetStandardOutput())));
         }

--- a/src/Cake.Common/Tools/DotNet/MSBuild/MSBuildArgumentBuilderExtensions.cs
+++ b/src/Cake.Common/Tools/DotNet/MSBuild/MSBuildArgumentBuilderExtensions.cs
@@ -1,4 +1,4 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
@@ -40,6 +40,12 @@ namespace Cake.Common.Tools.DotNet.MSBuild
             ArgumentNullException.ThrowIfNull(builder);
 
             var msBuilder = new ProcessArgumentBuilder();
+
+            // Verbosity: use MSBuild /verbosity switch (dotnet msbuild does not support --verbosity; it forwards to MSBuild which expects /verbosity:x)
+            if (settings.Verbosity.HasValue)
+            {
+                msBuilder.AppendMSBuildSwitch("verbosity", GetMSBuildVerbosityValue(settings.Verbosity.Value));
+            }
 
             // Got any targets?
             if (settings.Targets.Any())
@@ -287,6 +293,20 @@ namespace Cake.Common.Tools.DotNet.MSBuild
             var counter = index == 0 ? string.Empty : index.ToString();
             return $"/fileLogger{counter} /fileloggerparameters{counter}:{parameters}";
         }
+
+        /// <summary>
+        /// Maps <see cref="DotNet.DotNetVerbosity"/> to MSBuild verbosity string (quiet, minimal, normal, detailed, diagnostic).
+        /// </summary>
+        private static string GetMSBuildVerbosityValue(DotNet.DotNetVerbosity verbosity)
+            => verbosity switch
+            {
+                DotNet.DotNetVerbosity.Quiet => "quiet",
+                DotNet.DotNetVerbosity.Minimal => "minimal",
+                DotNet.DotNetVerbosity.Normal => "normal",
+                DotNet.DotNetVerbosity.Detailed => "detailed",
+                DotNet.DotNetVerbosity.Diagnostic => "diagnostic",
+                _ => throw new ArgumentOutOfRangeException(nameof(verbosity), verbosity, "Invalid value"),
+            };
 
         private static string GetToolVersionValue(MSBuildVersion toolVersion)
         {

--- a/tests/integration/Cake.Common/Tools/DotNet/DotNetAliases.cake
+++ b/tests/integration/Cake.Common/Tools/DotNet/DotNetAliases.cake
@@ -216,8 +216,8 @@ Task("Cake.Common.Tools.DotNet.DotNetAliases.DotNetMSBuild")
     var project = path.CombineWithFilePath("hwapp/hwapp.csproj");
     var assembly = path.CombineWithFilePath("hwapp/bin/Debug/net10.0/hwapp.dll");
 
-    // When
-    DotNetMSBuild(project.FullPath);
+    // When (Verbosity.Quiet exercises MSBuild /verbosity, not dotnet --verbosity; see https://github.com/cake-build/cake/issues/4456)
+    DotNetMSBuild(project.FullPath, new DotNetMSBuildSettings { Verbosity = DotNetVerbosity.Quiet });
 
     // Then
     Assert.True(System.IO.File.Exists(assembly.FullPath));

